src/Catel.MVVM/Catel.MVVM.Shared/Collections/SuspensionContext.cs

```diff
@@ -35,167 +26,46 @@ public class SuspensionContext<T>
         /// </summary>
-        public List<T> NewItems
-        {
-            get
-            {
-                return _newItems;
-            }
-        }
+        public List<int> ChangedItemIndices { get; } = new List<int>();
 
         /// <summary>
-        /// Gets the indices od the removed items while change notifications were suspended.
+        /// Gets the changed items while change notifications.
         /// </summary>
-        public List<int> OldItemIndices
-        {
-            get
-            {
-                return _oldItemIndices;
-            }
-        }
+        public List<T> ChangedItems { get; } = new List<T>();
 
         /// <summary>
-        /// Gets the removed items while change notifications were suspended.
+        /// Gets or sets the suspension count.
         /// </summary>
-        public List<T> OldItems
+        public int Count
         {
-            get
-            {
-                return _oldItems;
-            }
-        }
-        #endregion
+            get => _suspensionCount;
 
-        #region Methods
-        #endregion
-
-        /// <summary>
-        /// Tries to remove the item from old items
-        /// </summary>
-        /// <param name="index">
-        /// The item index.
-        /// </param>
-        /// <param name="item">
-        /// The item.
-        /// </param>
-        /// <returns>
-        /// <c>true</c> if removed, otherwise <c>false</c>.
-        /// </returns>
-        public bool TryRemoveItemFromOldItems(int index, T item)
-        {
-            if (Mode == SuspensionMode.None || Mode == SuspensionMode.Mixed)
+            set
             {
-                var oldIdx = OldItems.LastIndexOf(item);
-                if (oldIdx > -1 && OldItemIndices[oldIdx] == index)
+                if (value != _suspensionCount)
                 {
-                    OldItems.RemoveAt(oldIdx);
-                    OldItemIndices.RemoveAt(oldIdx);
-
-                    for (var i = 0; i < OldItemIndices.Count; i++)
-                    {
-                        if (OldItemIndices[i] >= index)
-                        {
-                            OldItemIndices[i]++;
-                        }
-                    }
-
-                    return true;
+                    _suspensionCount = value < 0 ? 0 : value;
                 }
             }
-
-            return false;
         }
 
         /// <summary>
-        /// Tries to remove the item from new items
+        /// Gets the actions while change notifications were suspended in Mixed mode.
         /// </summary>
-        /// <param name="index">
-        /// The item index.
-        /// </param>
-        /// <param name="item">
-        /// The item.
-        /// </param>
-        /// <returns>
-        /// <c>true</c> if removed, otherwise <c>false</c>.
-        /// </returns>
-        public bool? TryRemoveItemFromNewItems(int index, T item)
```

Comment:
Only FastBindingList is using this. So it was moved to ExtendedSuspensionContext.